openslides_backend/action/actions/option/update.py

```diff
@@ -0,0 +1,173 @@
+from ....shared.exceptions import ActionException
+from ....shared.patterns import Collection, FullQualifiedId
+from ...generics.update import UpdateAction
+from ...util.default_schema import DefaultSchema
+from ...util.register import register_action
+from ..vote.create import VoteCreate
+from ..vote.update import VoteUpdate
+
+
+@register_action("option.update")
+class OptionUpdateAction(UpdateAction):
+    """
+    Action to update a user.
+    """
+
+    model = Option()
+    schema = DefaultSchema(Option()).get_update_schema(
+        additional_optional_fields={
+            "Y": {"type": "string", "pattern": r"^-?(\d|[1-9]\d+)\.\d{6}$"},
+            "N": {"type": "string", "pattern": r"^-?(\d|[1-9]\d+)\.\d{6}$"},
+            "A": {"type": "string", "pattern": r"^-?(\d|[1-9]\d+)\.\d{6}$"},
+        }
+    )
+
+    def update_instance(self, instance: Dict[str, Any]) -> Dict[str, Any]:
+        """Update votes and auto calculate yes, no, abstain."""
+
+        poll_id_option, poll, option = self._get_poll(instance["id"])
+        if poll_id_option:
+            self._handle_poll_option_data(instance, poll)
+        else:
+            self._handle_global_option_data(instance, poll)
+
+        id_to_vote = self._fetch_votes(option.get("vote_ids", []))
+
+        payload_create = []
+        payload_update = []
+        if "yes" in instance:
+            vote_id = self._get_vote_id("Y", id_to_vote)
+            if vote_id is None:
+                payload_create.append(
+                    {
+                        "option_id": instance["id"],
+                        "value": "Y",
+                        "weight": instance["yes"],
+                        "meeting_id": option["meeting_id"],
+                    }
+                )
+            else:
+                payload_update.append({"id": vote_id, "weight": instance["yes"]})
+        if "no" in instance:
+            vote_id = self._get_vote_id("N", id_to_vote)
+            if vote_id is None:
+                payload_create.append(
+                    {
+                        "option_id": instance["id"],
+                        "value": "N",
+                        "weight": instance["no"],
+                        "meeting_id": option["meeting_id"],
+                    }
+                )
+            else:
+                payload_update.append({"id": vote_id, "weight": instance["no"]})
+        if "abstain" in instance:
+            vote_id = self._get_vote_id("A", id_to_vote)
+            if vote_id is None:
+                payload_create.append(
+                    {
+                        "option_id": instance["id"],
+                        "value": "A",
+                        "weight": instance["abstain"],
+                        "meeting_id": option["meeting_id"],
+                    }
+                )
+            else:
+                payload_update.append({"id": vote_id, "weight": instance["abstain"]})
```

Comment:
You can reduce the amount of redundant code by iterating over the needed values (`yes`, `no`, `abstain`)

Reply:
Use a loop to simplify the code here.